src/renderer/components/wallet/settings/WalletSettings.tsx

```diff
@@ -162,12 +169,46 @@ export const WalletSettings: React.FC<Props> = (props): JSX.Element => {
     ]
   )
 
+  const [verifyAddressModalVisible, setVerifyAddressModalVisible] = useState(false)
+  const [addressToVerify, setAddressToVerify] = useState('')
```

Comment:
You can check `addressToVerify` to open modal or not, don't you? No need for another state `verifyAddressModalVisible` then.